### PR TITLE
FEATURE: MSF-20653 Update message error blob storage

### PR DESCRIPTION
**Pull request by tqtu**

### CODE REVIEW

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/771/ : SUCCESS in 6s

**Comment by tqtu** [comment]
ok to test

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/772/ : SUCCESS in 9s

File: lib/gooddata/cloud_resources/blobstorage/blobstorage_client.rb [hunk 1]
```diff
@@ -59,5 +72,17 @@ def build_sas(url)
...
+
+    def raise_error(e)
+      if e.message && e.message.include?(INVALID_BLOB_EXPIRED_ORIGINAL_MESSAGE)
+        raise INVALID_BLOB_EXPIRED_MESSAGE
+      elsif e.message && e.message.include?(INVALID_BLOB_SIG_WELL_FORMED_MESSAGE)
+        raise INVALID_BLOB_SIG_WELL_FORMED_MESSAGE
+      elsif e.message && e.message.include?(INVALID_BLOB_CONTAINER_MESSAGE)
+        raise INVALID_BLOB_CONTAINER_FORMED_MESSAGE
+      else
+        raise INVALID_BLOB_GENERAL_MESSAGE
```
Comment (phong-nguyen-duy) [inline lib/gooddata/cloud_resources/blobstorage/blobstorage_client.rb:84]: how about the path is not found

File: lib/gooddata/cloud_resources/blobstorage/blobstorage_client.rb [hunk 2]
```diff
@@ -30,13 +38,18 @@ def initialize(options = {})
...
     def realize_blob(file, _params)
       GoodData.gd_logger.info("Realizing download from Blob Storage. Container #{@container}.")
-      connect
-      filename = "#{SecureRandom.urlsafe_base64(6)}_#{Time.now.to_i}.csv"
-      blob_name = @path ? "#{file}" : "#{@path.delete_suffix('/')}/#{file}"
+      filename = ''
+      begin
+        connect
+        filename = "#{SecureRandom.urlsafe_base64(6)}_#{Time.now.to_i}.csv"
+        blob_name = @path ? "#{file}" : "#{@path.delete_suffix('/')}/#{file}"
```
Comment (phong-nguyen-duy) [inline lib/gooddata/cloud_resources/blobstorage/blobstorage_client.rb:45]: is it reversed?

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/775/ : SUCCESS in 6s

File: lib/gooddata/cloud_resources/blobstorage/blobstorage_client.rb (hunk 1 above)
Comment (tqtu) [reply lib/gooddata/cloud_resources/blobstorage/blobstorage_client.rb:84]: With  wrong path, the exception was not thrown, and the downloaded file is empty. No errors
So It is not rethrow

File: lib/gooddata/cloud_resources/blobstorage/blobstorage_client.rb (hunk 2 above)
Comment (tqtu) [reply lib/gooddata/cloud_resources/blobstorage/blobstorage_client.rb]: Yes, the logic code is reverse
I fixed its.

**Comment by gdgate** [comment]
Build succeeded (check pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-oss-check/776/ : SUCCESS in 7s

File: lib/gooddata/cloud_resources/blobstorage/blobstorage_client.rb (hunk 1 above)
Comment (tqtu) [reply lib/gooddata/cloud_resources/blobstorage/blobstorage_client.rb:88]: With the wrong pass, BlobNotFound message will be thrown,
I fixed and rethrow to path not found

**Comment by yenkins** [comment]
### Sonar scan result


More detail, see in https://sonarqube-gate.intgdc.com/dashboard?id=gooddata-ruby-gate-PR1786

To scan for vulnerabilities in dependencies and run unit tests (to get coverage report in sonar) please comment your PR with 'extended check sonar'.

**Comment by gdgate** [comment]
Build succeeded (gate pipeline).

- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-merge-compliance-check/561/ : SUCCESS in 17s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-sonarqube-scan-zuul-gate/21/ : SUCCESS in 1m 32s
- https://checklist.intgdc.com/job/compliance/job/gooddata-ruby-fossa-scanning-zuul-el7/106/ : SUCCESS in 1m 10s (non-voting)

**Pull request merged**